packages/styled-components/src/models/StyledComponent.js

```diff
@@ -238,20 +230,23 @@ export default function createStyledComponent(
     isTargetStyledComp ? ((target: Object).componentStyle: ComponentStyle) : undefined
   );
 
+  // statically styled-components don't need to build an execution context object,
+  // and shouldn't be increasing the number of class names
+  const isStatic = componentStyle.isStatic && attrs.length === 0;
```

Comment:
Hoisted this up to the definition layer rather than runtime layer so we can use it for future optimizations...